### PR TITLE
Document key feature icons

### PR DESCRIPTION
**Pull request by ErikSchierboom**
CC @ee7

### CODE REVIEW

**Review by iHiD — APPROVED** [review]
I'll approve and you can merge in the morning if ee7 doesn't get chance to look before then :)

File: building/configlet/lint.md [hunk 1]
```diff
@@ -181,7 +181,7 @@ The `config.json` file should have the following checks:
 - The `"key_features"` key is optional
 - The `"key_features"` value must be an array with length = 6
 - The `"key_features[].icon"` key is required
-- The `"key_features[].icon"` value must be a string that matches one of the pre-defined icon values (TODO: add link to list of icons)
+- The `"key_features[].icon"` value must use one of the [pre-defined icon values](../tracks/config-json.md#keyfeatures)
```
Comment (ee7) [inline building/configlet/lint.md]:
```suggestion
- The `"key_features[].icon"` value must use one of the [pre-defined icon values](../tracks/config-json#keyfeatures)
```

IIRC this should this omit the `.md` extension. (There are others in this file, but I'll eventually fix them in https://github.com/exercism/docs/pull/160).

File: building/tracks/config-json.md [hunk 2]
```diff
@@ -435,32 +461,32 @@ This is an example of what a valid `config.json` file can look like:
...
   "key_features": [
     {
-      "icon": "features-oop",
+      "icon": "expressive",
       "title": "Modern",
       "content": "C# is a modern, fast-evolving language."
     },
     {
-      "icon": "features-strongly-typed",
+      "icon": "statically-typed",
```
Comment (ee7) [inline building/tracks/config-json.md]:
```suggestion
      "icon": "cross-platform",
```

We want to better match the below, right?
```

      "title": "Cross-platform",
      "content": "C# runs on almost any platform and chipset."
```

File: building/tracks/config-json.md [hunk 3]
```diff
@@ -435,32 +461,32 @@ This is an example of what a valid `config.json` file can look like:
...
       "title": "Cross-platform",
       "content": "C# runs on almost any platform and chipset."
     },
     {
-      "icon": "features-functional",
+      "icon": "functional",
```
Comment (ee7) [inline building/tracks/config-json.md]:
```suggestion
      "icon": "multi-paradigm",
```

Assuming that we want to better match the below
```
      "title": "Multi-paradigm",
      "content": "C# is primarily an object-oriented language, but also has lots of functional features."
```

(unless the point was to show that we can choose the icon independently).

File: building/tracks/config-json.md [hunk 4]
```diff
@@ -249,7 +249,33 @@ The key features are specified in the top-level `key_features` field which is de
...
+  - `documentation`
+  - `dynamically-typed`
+  - `easy`
+  - `embeddable`
+  - `expressive`
+  - `extensible`
+  - `fast`
+  - `fun`
+  - `functional`
+  - `garbage`
```
Comment (ee7) [inline building/tracks/config-json.md]:
```suggestion
  - `garbage-collected`
```

Better? `garbage` alone will just sound like "bad" to many ears.

Edit: I realize that this should match the filename in [exercism/website-icons](https://github.com/exercism/website-icons/tree/main/key-features). However, the other icons are named for the feature, not the image (e.g. we have `scientific`, not `flask`. And `fast`, not `rabbit`.)

File: building/tracks/config-json.md [hunk 5]
```diff
@@ -249,7 +249,33 @@ The key features are specified in the top-level `key_features` field which is de
...
+  - `homoiconic`
+  - `interactive`
+  - `interop`
+  - `multi-paradigm`
+  - `portable`
+  - `powerful`
+  - `productive`
+  - `safe`
+  - `scientific`
+  - `shrink`
```
Comment (ee7) [inline building/tracks/config-json.md]: What is `shrink`?

Edit: I see that it looks like https://raw.githubusercontent.com/exercism/website-icons/main/key-features/shrink.svg

File: building/tracks/config-json.md [hunk 6]
```diff
@@ -259,9 +285,9 @@ Exactly 6 key features must be specified.
 {
   "key_features": [
     {
-      "title": "Modern",
-      "content": "C# is a modern, fast-evolving language.",
-      "icon": "features-oop"
+      "title": "Fault tolerant",
```
Comment (ee7) [inline building/tracks/config-json.md]:
```suggestion
      "title": "Fault-tolerant",
```

File: building/tracks/config-json.md (hunk 5 above)
Comment (ErikSchierboom) [reply building/tracks/config-json.md]: I think that's the icon for a language producing small binanies. @taiyab?

edit: it is

File: building/tracks/config-json.md (hunk 4 above)
Comment (ErikSchierboom) [reply building/tracks/config-json.md]: It makes sense to rename the icon to `garbage-collected`. I'll do that.

Comment (ErikSchierboom) [reply building/tracks/config-json.md]: Done: https://github.com/exercism/website-icons/pull/71/files

**Comment by ErikSchierboom** [comment]
@ee7 Updated.

**Review by ee7 — APPROVED** [review]
Nice. I was going to ask if `shrink` could be `small` :)

**Pull request merged**